src/SqlBinding/TriggerBinding/SqlChangeTrackingConverter.cs

```diff
@@ -0,0 +1,181 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Data.SqlClient;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Sql
+{
+    internal class SqlChangeTrackingConverter
+    {
+        private readonly string _table;
+        private readonly string _connectionStringSetting;
+        private readonly IConfiguration _configuration;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SqlChangeTrackingConverter"/> class.
+        /// </summary>
+        /// <param name="connectionStringSetting"> 
+        /// The name of the app setting that stores the SQL connection string
+        /// </param>
+        /// <param name="table"> 
+        /// The name of the user table that changes are being tracked on
+        /// </param>
+        /// <param name="configuration">
+        /// Used to extract the connection string from connectionStringSetting
+        /// </param>
+        /// <exception cref="ArgumentNullException">
+        /// Thrown if any of the parameters are null
+        /// </exception>
+        public SqlChangeTrackingConverter(string table, string connectionStringSetting, IConfiguration configuration)
+        {
+            _table = table ?? throw new ArgumentNullException(nameof(table));
+            _connectionStringSetting = connectionStringSetting ?? throw new ArgumentNullException(nameof(connectionStringSetting));
+            _configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
+        }
+
+        /// <summary>
+        /// Returns a list of SqlChangeTrackingEntry. Each entry is populated with the type of the change
+        /// and the associated data from the user table for each row in rows
+        /// </summary>
+        /// <typeparam name="T">
+        /// The POCO representing a row of the user's table
+        /// </typeparam>
+        /// <param name="rows">
+        /// The list of rows that were changed. Each row is populated by its primary key values,
+        /// as well as all columns from both the worker table and change table (<see cref="ChangeTableData.workerTableRows"/>)
+        /// </param>
+        /// <param name="whereChecksOfRows">
+        /// Used to build up the query to read the associated row from the user table (<see cref="ChangeTableData.whereChecksOfRows"/>)
+        /// </param>
+        /// <returns></returns>
+        public async Task<object> BuildSqlChangeTrackingEntries<T>(List<Dictionary<String, String>> rows, 
```

Comment:
I need to cast the result of this method call [here](https://github.com/Azure/azure-functions-sql-extension/blob/triggerbinding/src/SqlBinding/TriggerBinding/SqlTriggerBinding.cs#L191) to be a Task so that I can await it. I can't actually cast it to `Task<IEnumerable<SqlChangeTrackingEntry<T>>>` since I don't know what T is until runtime. And I also can't cast it to `Task<object>` if the return type of `BuildSqlChangeTrackingEntries` is  `Task<IEnumerable<SqlChangeTrackingEntry<T>>>` (I tried). So this seemed to be the only way I could get it to work.